src/overlays/actors/ovl_En_Po_Fusen/z_en_po_fusen.c

```diff
@@ -21,34 +29,232 @@ const ActorInit En_Po_Fusen_InitVars = {
+       this->actor.world.pos.x, this->actor.world.pos.y + 20.0f, this->actor.world.pos.z, 
+       255, 255, 200, 2);
+    func_800B8EC8(&this->actor, 0x180E); // NA_SE_IT_BOMB_EXPLOSION sfx
+    Actor_MarkForDeath(&this->actor);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Po_Fusen_0x80AC1270/func_80AC18EC.asm")
+void EnPoFusen_InitFuse(EnPoFusen *this) {
+    s16 rotZ = this->actor.shape.rot.z;
+    this->fuse = this->actor.params & 0x3FF;
+    this->actor.shape.rot.z = 0;
+    this->randScaleChange = rotZ & 0xFFFFu;
+    this->actionFunc = EnPoFusen_IdleFuse;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Po_Fusen_0x80AC1270/func_80AC192C.asm")
+void EnPoFusen_IdleFuse(EnPoFusen *this, GlobalContext *gCtx) {
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Po_Fusen_0x80AC1270/func_80AC19B0.asm")
+    EnPoFusen_Idle(this, gCtx);
+    if ((this->fuse--) == 0) {
+        EnPoFusen_IncrementMalonPop(this);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Po_Fusen_0x80AC1270/func_80AC19DC.asm")
+void EnPoFusen_Update(Actor *thisx, GlobalContext *globalCtx) {
+    EnPoFusen* this = THIS;
+    this->actionFunc(this, globalCtx);
+    if (EnPoFusen_CheckCollision(this, globalCtx) != 0) {
+        EnPoFusen_IncrementMalonPop(this);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Po_Fusen_0x80AC1270/EnPoFusen_Update.asm")
+s32 EnPoFusen_OverrideLimbDraw(GlobalContext *gCtx, s32 limbIndex, Gfx **dList, Vec3f *pos, Vec3s *rot, struct Actor *actor) {
+    EnPoFusen* this = (EnPoFusen*)actor;
+    f32 zScale;
+    f32 yScale;
+    f32 xScale;
+    s16 pad;
+    s16 zRot;
+    s16 pad2;
+    s16 xRot;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Po_Fusen_0x80AC1270/func_80AC1A68.asm")
+    if (limbIndex == 2) {
+        zScale = (Math_Coss(this->randScaleChange) * 0.0799999982119f) + 1.0f;
+        xScale = zScale;
+        if (!zScale) { }
+        yScale = (Math_Sins(this->randScaleChange) * 0.0799999982119f) + 1.0f;
+        yScale = yScale * yScale;
+        xRot = (s16) ((Math_Sins(this->randXZRotChange) * 2730.0f));
+        zRot = (s16) ((Math_Coss(this->randXZRotChange) * 2730.0f));
+        SysMatrix_InsertRotation(xRot, 0, zRot , 1);
+        SysMatrix_InsertScale(xScale, yScale , zScale, 1);
+        SysMatrix_InsertZRotation_s( -zRot, 1);
+        SysMatrix_InsertXRotation_s( -xRot, 1);
+    } else if (limbIndex == 3) {
+        rot->y += this->limb3Rot;
+    } else if (limbIndex == 6) {
+        rot->y += this->limb46Rot;
+    } else if (limbIndex == 4) {
+        rot->z += this->limb46Rot;
+    } else if ((limbIndex == 5) || (limbIndex == 7)) {
+        rot->z += this->limb57Rot;
+    } else if (limbIndex == 8) {
+        rot->z += this->limb8Rot;
+    } else if (limbIndex == 9) {
+        rot->y += (s16) ( this->limb9Rot * Math_Sins(this->randBaseRotChange));
+        rot->z += (s16) ( this->limb9Rot * Math_Coss(this->randBaseRotChange));
```

Comment:
Remove casts

Reply:
![req casts](https://user-images.githubusercontent.com/42048411/111395085-3eefcd80-8679-11eb-99df-7977fdf91d0f.png)
These two are required, removing the rest that aren't.